src/gt4py/_core/locking.py

```diff
@@ -0,0 +1,26 @@
+# GT4Py - GridTools Framework
+#
+# Copyright (c) 2014-2024, ETH Zurich
+# All rights reserved.
+#
+# Please, refer to the LICENSE file in the root directory.
+# SPDX-License-Identifier: BSD-3-Clause
+
+from __future__ import annotations
+
+import pathlib
+from typing import ContextManager, Protocol
+
+import filelock
+
+
+class _LockConstructor(Protocol):
+    def __call__(self, path: str, /) -> ContextManager: ...
+
+
+_default_lock: _LockConstructor = filelock.FileLock
+
+
+def lock(path: pathlib.Path | str) -> ContextManager:
```

Comment:
Another option  be to pass here the lock_constructor as as keyword argument with a default set to `_default_lock`, right? Or do do you think it could error prone?

```suggestion
def lock(path: pathlib.Path | str, *, lock_constructor: _LockConstructor = _default_lock) -> ContextManager:
```

Reply:
What advantage do you see? Probably it would be simpler to directly use the custom lock?
I mean why
```
with gt4py.lock(path, lock_constructor=mylock):
    ...
```
over 
```
with mylock(path):
    ...
```

Reply:
I think you're right and my suggestion doesn't have any advantage. Thinking about it, I got distracted by the `_default_lock` name since it's not really a default but the __only__ lock constructor. Woud it be better to rename it to `_lock_constructor`?